src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -8,55 +8,76 @@
 import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
 import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
 import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Commands;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.Constants.RobotConstants;
+import frc.robot.commands.ElevatorCommands;
 import frc.robot.parameters.ElevatorLevel;
 import frc.robot.util.LimitSwitch;
 import frc.robot.util.MotorController;
 import frc.robot.util.RelativeEncoder;
 import frc.robot.util.TalonFXAdapter;
+import java.util.Set;
 
-public class Elevator extends SubsystemBase {
+@RobotPreferencesLayout(groupName = "Elevator", row = 1, column = 0, width = 1, height = 2)
+public class Elevator extends SubsystemBase implements ShuffleboardProducer {
+  @RobotPreferencesValue
+  public static RobotPreferences.BooleanValue ENABLE_TAB =
+      new RobotPreferences.BooleanValue("Elevator", "Enable Tab", false);
 
   // physical parameters of the elevator
-  private static final double GEAR_RATIO = (60.0 / 12.0) * (24.0 / 15.0);
-  private static final double SPROCKET_DIAMETER = 0.0475; // 4.75 cm
-  private static final double MASS = 2.0; // kilograms
+  private static final double GEAR_RATIO = ((60.0 / 12.0) * (24.0 / 15.0)) / 2;
+  private static final double SPROCKET_DIAMETER = 0.05; // 5 cm
+  private static final double MASS = 1.5; // kilograms
   private static final double METERS_PER_REVOLUTION = (SPROCKET_DIAMETER * Math.PI) / GEAR_RATIO;
 
+  private static final double MAX_HEIGHT = 1.42;
+  public static final double MIN_HEIGHT = 0.033; // in meters
+  private static final double DISABLE_HEIGHT = MIN_HEIGHT + 0.04;
+
   // trapezoid profile values
   private static final DCMotor MOTOR_PARAMS = DCMotor.getKrakenX60(1);
   private static final double MAX_SPEED =
-      (MOTOR_PARAMS.freeSpeedRadPerSec * SPROCKET_DIAMETER) / (GEAR_RATIO * 2 * Math.PI); // m/s
+      (MOTOR_PARAMS.freeSpeedRadPerSec / (2 * Math.PI)) * METERS_PER_REVOLUTION; // m/s
   private static final double MAX_ACCELERATION =
-      (2 * MOTOR_PARAMS.stallTorqueNewtonMeters * GEAR_RATIO) / (SPROCKET_DIAMETER * MASS); // m/s^2
+      (2 * MOTOR_PARAMS.stallTorqueNewtonMeters * GEAR_RATIO)
+          / (SPROCKET_DIAMETER * MASS); // m/s^2 for two motors
   private static final TrapezoidProfile.Constraints CONSTRAINTS =
-      new TrapezoidProfile.Constraints(MAX_SPEED, MAX_ACCELERATION);
+      new TrapezoidProfile.Constraints(MAX_SPEED / 2, MAX_ACCELERATION / 8);
 
   // feedforward constants
-  private static final double KS = 0.15;
-  private static final double KV = 12 / MAX_SPEED;
-  private static final double KA = 12 / MAX_ACCELERATION;
+  private static final double KS = 0.0656;
+  private static final double KV = (12 - KS) / MAX_SPEED;
+  private static final double KA = (12 - KS) / MAX_ACCELERATION;
```

Comment:
Let's use `RobotConstants.MAX_BATTERY_VOLTAGE` here.
```suggestion
  private static final double KV = (RobotConstants.MAX_BATTERY_VOLTAGE- KS) / MAX_SPEED;
  private static final double KA = (RobotConstants.MAX_BATTERY_VOLTAGE- KS) / MAX_ACCELERATION;
```